File: lib/layer2_core/dark/e_on_secondary_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class EOnSecondaryDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd26;
      case 'standard': return cstd26;
      case 'blue': return cbd26;
      case 'green': return cgd26;
      case 'red': return crd26;
      case 'nothing': return cnod26;
      default: return cstd26;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EOnSecondaryDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnSecondaryDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd26
            AppTheme.STANDARD -> R.color.cstd26
            AppTheme.BLUE -> R.color.cbd26
            AppTheme.GREEN -> R.color.cgd26
            AppTheme.RED -> R.color.crd26
            AppTheme.NOTHING -> R.color.cnod26
        }
    }
}
